spa/src/locale.ts

```diff
@@ -0,0 +1,79 @@
+import ICU from "i18next-icu";
+import i18nextXhrBackend from "i18next-xhr-backend";
+import LanguageDetector from "i18next-browser-languagedetector";
+
+const languageChangeObserver = new MutationObserver(() => {
+  const reDetect: any = undefined;
+  window.i18next.changeLanguage(reDetect).catch((e) => {
+    console.error("i18next failed to re-detect language");
+    console.error(e);
+  });
+});
+
+languageChangeObserver.observe(document.documentElement, {
+  attributeFilter: ["lang"],
+  attributes: true,
+});
+
+function decodeHtmlEntity(html: string) {
+  const textArea = document.createElement("textarea");
+  textArea.innerHTML = html;
+  return textArea.value;
+}
+
+export function setupLocale() {
```

Comment:
I think the old name was a clearer description of what this actually does. A locale is a string like "en-GB" — it's not obvious how one "sets one up." i18next is the thing being set up. If for some reason you want to lift the name one level of abstraction, you could call it "i18n" instead of "i18next."